acceptance/detector_test.go

```diff
@@ -0,0 +1,197 @@
+// +build acceptance
+
+package acceptance
+
+import (
+	"io/ioutil"
+	"math/rand"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"testing"
+	"time"
+
+	"github.com/BurntSushi/toml"
+	"github.com/sclevine/spec"
+	"github.com/sclevine/spec/report"
+
+	"github.com/buildpacks/lifecycle"
+	h "github.com/buildpacks/lifecycle/testhelpers"
+)
+
+var (
+	detectDockerContext = filepath.Join("testdata", "detector")
+	detectorBinaryDir   = filepath.Join("testdata", "detector", "container", "cnb", "lifecycle")
+	detectImage         = "lifecycle/acceptance/detector"
+	userID              = "1234"
+)
+
+func TestDetector(t *testing.T) {
+	h.SkipIf(t, runtime.GOOS == "windows", "Detector acceptance tests are not yet supported on Windows")
+
+	rand.Seed(time.Now().UTC().UnixNano())
+
+	h.MakeAndCopyLifecycle(t, "linux", detectorBinaryDir)
+	h.DockerBuild(t, detectImage, detectDockerContext)
+	defer h.DockerImageRemove(t, detectImage)
+
+	spec.Run(t, "acceptance-detector", testDetector, spec.Parallel(), spec.Report(report.Terminal{}))
+}
+
+func testDetector(t *testing.T, when spec.G, it spec.S) {
+	when("called with arguments", func() {
+		it("errors", func() {
+			cmd := exec.Command("docker", "run", "--rm", detectImage, "some-arg")
+			output, err := cmd.CombinedOutput()
+			h.AssertNotNil(t, err)
+			expected := "failed to parse arguments: received unexpected arguments"
+			h.AssertStringContains(t, string(output), expected)
+		})
+	})
+
+	when("running as a root", func() {
+		it("errors", func() {
+			cmd := exec.Command("docker", "run", "--rm", "--user", "root", detectImage)
+			output, err := cmd.CombinedOutput()
+			h.AssertNotNil(t, err)
+			expected := "failed to build: refusing to run as root"
+			h.AssertStringContains(t, string(output), expected)
+		})
+	})
+
+	when("read buildpack order file failed", func() {
+		it("errors", func() {
+			// no order.toml file in the default directory
+			cmd := exec.Command("docker", "run", "--rm", "--user", userID, detectImage)
```

Comment:
It might be worth setting `USER ${cnb_uid}:${cnb_gid}` in the detect image `Dockerfile` so we don't need to remember to pass `--user` in every test, unless the test is explicitly testing user related logic(e.g. `--user root` in the test above).